lib/trackler/problem.rb

```diff
@@ -7,81 +7,108 @@ class Problem
     def initialize(slug, root)
       @slug = slug
       @root = root
+      @xcommon_repository_url = 'https://github.com/exercism/x-common/blob/master'
     end
 
     def exists?
-      !!md && !!yaml
+      description_exists? && yaml_exists?
     end
 
     def name
-      @name ||= slug.split('-').map(&:capitalize).join(' ')
+      @name ||= slug_to_name
     end
 
     def description
-      @description ||= File.read(filepath(md))
+      @description ||= load_description
     end
 
     def source_markdown
-      body = source.empty? ? "" : "%s" % source
-      url = source_url.empty? ? "" : "[%s](%s)" % [source_url, source_url]
-      if url.empty? && body.empty?
-        ""
-      else
-        "## Source\n\n" + [body, url].reject(&:empty?).join(" ")
-      end
+      text = [source, markdown_link(source_url)].reject(&:empty?).join(" ")
+      text.empty? ? text : "## Source\n\n#{text}"
     end
 
     def md_url
-      repo_url(md)
+      repo_url(md_path)
     end
 
     def json_url
-      repo_url(json) if !!json
+      repo_url(json_path) if json_path
```

Comment:
> json_path isn't a boolean.

Why does that matter? It's either `truthy` or `falsey` which is all the `if` cares about.
Is there a case where these 2 statements are not equivalent?

I could see arguments for either `repo_url(json_path) unless json_path.nil?`
or `repo_url(json_path)` (and letting repo_url deal with nil values) being better.